eip-0017.md

```diff
@@ -0,0 +1,70 @@
+# Proxy contracts
+
+* Author: anon_real
+* Status: Proposed
+* Created: 05-May-2021
+* License: CC0
+* Forking: not needed 
+
+## Motivation 
+
+Outsourcing transaction generation to an external service/dApp can be useful or needed in various circumstances. For example, avoiding wallet limitations to generate any transaction on behalf of the user - [Ergo Assembler](https://github.com/anon-real/ergo-assembler) is designed for this purpose. Another example is to scale dApps to be able to fulfill many requests without double-spending or data invalidation - SigmaUSD dApp can use proxy contracts to avoid bank double-spending and ERG/USD oracle data invalidation.
+
+
+## Background
+The idea of proxy contracts came to life with the [Ergo Assembler](https://github.com/anon-real/ergo-assembler) which helped dApp developments like [Ergo Auction House](https://ergoauctions.org/), [ErgoUtils](https://ergoutils.org/), and [SigmaUSD web interface](https://sigmausd.io/#/) despite not having a wallet-bridge like MetaMask (Ethereum wallet) in the ecosystem.
+
+During this time, the structure of proxy contracts evolved as some malicious users tried to take advantage of some minor vulnerabilities, mostly in the [SigmaUSD dApp](https://sigmausd.io/#/).
+
+## The structure
+In the beginning, the sole purpose of proxy contracts was to protect users from losing their funds (not to be cheated) when they outsource their assets to engage with some dApp. While the initial structure succeeded to achieve this, it proved to be not sufficient for the whole dApp infrastructure to work without malicious activities. 
+
+For example, a malicious whale tried to take advantage of this simple structure by stealing UI fees from SigmaUSD web interface developers for some period of time. This happened because the proxy contracts were simply only trying to protect users from malicious activities, not the dApp infrastructure.
+
+The same whale tried to prevent user's requests (minting/redeeming) from being executed by the assemble service by retuning their funds as soon as funds were broadcasted in the network. This happened also because of the same reasons.
+
+Generally, proxy contracts should be designed to:
+- prevent dApp developers or any other attacker from taking advantage of user's funds in any manner
+- preserve the integrity of the dApp by preventing attacks like the ones explained in the above examples.
+
+To achieve all of the above, the below contract structure is proposed as an example:
+
```

Comment:
The aim of the EIP is to propose a structure for proxy contracts and discuss some aspects/attacks for them. It does so with the SigmaUSD proxy contract because has evolved a lot through time with the whale's attacks. I think the discussed elements are useful for any proxy contracts (like for DEX).
If you think one proxy contract is not enough, I can add more.